File: Solutions in Python/p173.py
# 
# Solution to Project Euler problem 173
# Copyright (c) Sankha Saha (CODR). All rights reserved.
# 
#
# https://github.com/codr07/Project-Euler-Solusions
# 


def compute():
	TILES = 10**6
	ans = 0
	for n in range(3, TILES // 4 + 2):  # Outer square length
		for k in range(n - 2, 0, -2):  # Inner square length
			if n * n - k * k > TILES:
				break
			ans += 1
	return str(ans)


if __name__ == "__main__":
	print(compute())

File: Solutions In Java/p173.java
/* 
 * Solution to Project Euler problem 173
 * Copyright (c) Sankha Saha (CODR). All rights reserved.
 * 
 * https://github.com/codr07/Project-Euler-Solusions
 */


public final class p173 implements EulerSolution {
	
	public static void main(String[] args) {
		System.out.println(new p173().run());
	}
	
	
	private static final int TILES = 1000000;
	
	public String run() {
		int count = 0;
		for (int n = 3; n <= TILES / 4 + 1; n++) {  // Outer square length
			for (int k = n - 2; k >= 1; k -= 2) {   // Inner square length
				if ((long)n * n - (long)k * k > TILES)
					break;
				count++;
			}
		}
		return Integer.toString(count);
	}
	
}
